src/hiposfer/kamal/server.clj

```diff
@@ -38,10 +50,16 @@
                      {language :- string? "en"}]
       :return ::spec/direction
       (ok (let [coordinates (parse-coordinates coordinates)
-                radiuses    (some-> radiuses (parse-radiuses))]
-            (if (and (not-empty radiuses) (not= (count radiuses) (count coordinates)))
+                radiuses    (some-> radiuses (parse-radiuses))
+                out-of-range-coord (first (filter #(not(apply coord-in-range? %)) coordinates))]
```

Comment:
@mehdisadeghi kudos for the parsing and validation function :)

Nevertheless this can be greatly simplified with `clojure.spec`. In fact we already have a [spec for that](https://github.com/hiposfer/geojson.specs/blob/master/src/hiposfer/geojson/specs.clj#L33), but we simply forgot to validate the coordinates :D

I think something like this should do the job
```
(cond
  (s/valid? :hiposfer.geojson.specs.linestring/coordinates coordinates) (s/explain-data  :hiposfer.geojson.specs.linestring/coordinates coordinates}
.... do the rest
)
```

Reply:
I told you, you'll squeeze down everything into one line :)